pkg/ccl/streamingccl/streamproducer/producer_job.go

```diff
@@ -105,13 +131,73 @@ func doStartReplicationStream(
 		return streaming.InvalidStreamID, errors.New("admin role required to start stream replication jobs")
 	}
 
-	registry := evalCtx.ExecConfigAccessor.JobRegistry().(*jobs.Registry)
+	registry := execConfig.JobRegistry
 	timeout := streamingccl.StreamReplicationJobLivenessTimeout.Get(&evalCtx.Settings.SV)
-	jobID, jr := makeProducerJobRecord(registry, tenantID, timeout, evalCtx.SessionData().User())
-	if _, err := registry.CreateAdoptableJobWithTxn(evalCtx.Ctx(), jr, jobID, txn); err != nil {
+	ptsID := uuid.MakeV4()
+	jr := makeProducerJobRecord(registry, tenantID, timeout, evalCtx.SessionData().User(), ptsID)
+	if _, err := registry.CreateAdoptableJobWithTxn(evalCtx.Ctx(), jr, jr.JobID, txn); err != nil {
 		return streaming.InvalidStreamID, err
 	}
-	return streaming.StreamID(jobID), nil
+
+	ptp := execConfig.ProtectedTimestampProvider
+	statementTime := hlc.Timestamp{
+		WallTime: evalCtx.GetStmtTimestamp().UnixNano(),
+	}
+	pts := jobsprotectedts.MakeRecord(ptsID, int64(jr.JobID), statementTime,
+		[]roachpb.Span{*makeTenantSpan(tenantID)}, jobsprotectedts.Jobs)
+
+	if err := ptp.Protect(evalCtx.Ctx(), txn, pts); err != nil {
+		return streaming.InvalidStreamID, err
+	}
```

Comment:
I'm pretty sure we want to create PTS record before we start the job.  E.g. see changefeed_stmt.
Basically, we don't ever want to run streaming w/out knowing we have protected data.